subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskDefinitionIntegrationTest.groovy

```diff
@@ -478,4 +478,51 @@ class TaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         then:
         result.output.contains("got it 15")
     }
+
+    def "renders deprecation warning when adding a pre-created task to the task container"() {
+        given:
+        buildFile << """
+            Task foo = tasks.create("foo")
+            
+            tasks.add(new Bar("bar", foo))
+            
+            class Bar implements Task {
+                String name
+                
+                @Delegate
+                Task delegate
+                
+                Bar(String name, Task delegate) {
+                    this.name = name
+                    this.delegate = delegate
+                }
+                
+                String getName() {
+                    return name
+                }
+            }
+        """
+
+        when:
+        executer.expectDeprecationWarning()
+        succeeds("help")
+
+        then:
+        outputContains("The add() method has been deprecated.")
```

Comment:
should check that the rest of the message doesn't say "and is scheduled for removal in Gradle 5.0" because this method is inherited from `java.util.Collection` so we won't be removing it. It will instead become an error to call this method on a task container.

Reply:
Good point.  I've changed it.